### PR TITLE
Add a few group call event handler tests

### PR DESCRIPTION
**Pull request by SimonBrandner**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/test-utils/webrtc.ts [hunk 1]
```diff
@@ -421,15 +422,16 @@ export function installWebRTCMocks() {
     };
 }
 
-export function makeMockGroupCallStateEvent(roomId: string, groupCallId: string): MatrixEvent {
+export function makeMockGroupCallStateEvent(roomId: string, groupCallId: string, content: IContent = {
+    "m.type": GroupCallType.Video,
+    "m.intent": GroupCallIntent.Prompt,
+    "m.terminated": false,
```
Comment (robintown) [inline spec/test-utils/webrtc.ts:428]: According to MSC3401, the presence of the `m.terminated` field is supposed to be enough to mark the call as terminated, which I presume isn't what you were going for here. If the group call handler doesn't treat this as terminated, I think we should change it to do that

File: spec/unit/webrtc/groupCallEventHandler.spec.ts [hunk 2]
```diff
@@ -137,6 +183,42 @@ describe('Group Call Event Handler', function() {
...
+                    "m.intent": GroupCallIntent.Prompt,
+                    "dataChannelsEnabled": true,
+                    dataChannelOptions,
+                },
+            ),
+            {
+                roomId: FAKE_ROOM_ID,
+            } as unknown as RoomState,
+            null,
+        );
```
Comment (robintown) [inline spec/unit/webrtc/groupCallEventHandler.spec.ts:212]: Maybe this should be factored out into a method on MockCallMatrixClient, given that you're sending state events like this multiple times

File: spec/unit/webrtc/groupCallEventHandler.spec.ts [hunk 3]
```diff
@@ -137,6 +183,42 @@ describe('Group Call Event Handler', function() {
...
+
+        // @ts-ignore Mock
```
Comment (robintown) [inline spec/unit/webrtc/groupCallEventHandler.spec.ts:214]: It's not obvious what this is bypassing TS for

File: spec/unit/webrtc/groupCallEventHandler.spec.ts [hunk 4]
```diff
@@ -159,4 +241,86 @@ describe('Group Call Event Handler', function() {
...
+            expect(mockClientEmit).not.toHaveBeenCalledWith(
+                GroupCallEventHandlerEvent.Incoming,
+                expect.objectContaining({
+                    groupCallId: FAKE_GROUP_CALL_ID,
+                }),
+            );
+        });
+
+        it("ignores calls without a room", async () => {
+            (mockClient as any).getRoom = jest.fn().mockReturnValue(undefined);
```
Comment (robintown) [inline spec/unit/webrtc/groupCallEventHandler.spec.ts:314]: Is it possible to spy this instead?

File: spec/test-utils/webrtc.ts (hunk 1 above)
Comment (SimonBrandner) [reply spec/test-utils/webrtc.ts]: Argh, good point! (typing...)

**Review by robintown — APPROVED** [review]

**Pull request merged**